File: CookBook.Recipes/src/CookBook.Recipes.Api/Recipes/Endpoints/GetRecipeDetail/Contracts/RecipeDetailDto.cs
namespace CookBook.Recipes.Api.Recipes.Endpoints.GetRecipeDetail.Contracts;

internal sealed record RecipeDetailDto
{
    public required long RecipeId { get; init; }

    public required string UserName { get; init; }

    public required string Title { get; init; }

    public required string? Description { get; init; }

    public required short Servings { get; init; }

    public required short CookTime { get; init; }

    public required string? Notes { get; init; }

    public required DateTimeOffset CreatedAt { get; init; }

    public required IReadOnlyCollection<IngredientItemDto> Ingredients { get; init; }

    public required IReadOnlyCollection<InstructionItemDto> Instructions { get; init; }

    public required IReadOnlyCollection<string> Tags { get; init; }

    public sealed record IngredientItemDto
    {
        public required int LocalId { get; init; }

        public required string Note { get; init; }
    }

    public sealed record InstructionItemDto
    {
        public required int LocalId { get; init; }

        public required string Note { get; init; }
    }
}

File: CookBook.RecipesWebapp/client/src/api/recipes/dto/RecipeDetailDto.ts
export type RecipeDetailDto = {
  recipeId: number;
  title: string;
  description: string | null;
  servings: number;
  cookTime: number;
  notes: string | null;
  createdAt: string;
  userProfileInfo: UserProfileInfoDto;
  ingredients: RecipeDetailIngredientDto[];
  instructions: RecipeDetailInstructionDto[];
  tags: string[];
};

export type UserProfileInfoDto = {
  userName: string;
  displayName: string;
};

export type RecipeDetailIngredientDto = {
  localId: number;
  note: string;
};

export type RecipeDetailInstructionDto = {
  localId: number;
  note: string;
};
